crates/sui-source-validation/src/lib.rs

```diff
@@ -56,6 +58,30 @@ pub enum SourceVerificationError {
     InvalidModuleFailure { name: String, message: String },
 }
 
+#[derive(Debug)]
+pub struct AggregateSourceVerificationError(Vec<SourceVerificationError>);
+
+impl From<SourceVerificationError> for AggregateSourceVerificationError {
+    fn from(error: SourceVerificationError) -> Self {
+        AggregateSourceVerificationError(vec![error])
+    }
+}
+
+impl fmt::Display for AggregateSourceVerificationError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let AggregateSourceVerificationError(errors) = self;
+        for (i, error) in errors.iter().enumerate() {
+            if i != 0 {
+                write!(f, "\n\tNext error: ")?;
+            }
+            write!(f, "{}", error)?;
+        }
```

Comment:
Your thoughts here on preferred output format!

Reply:
I have a minor preference to treat all errors uniformly, if there are multiple, rather than the first error and then next, and next, etc, so something like:

```
Multiple source verification errors found:

- Error 1
- Error 2
- ...
```

But it makes sense to me that if there is only one error, then you only want to show that:


```rust
let AggregateSourceVerificationError(errors) = self;
match &errors[..] {
    [] => unreachable!("Aggregate error with no errors"),
    [error] => write!(f, "{}", error),
    errors => {
        write!(f, "Multiple source verification errors found:\n");
        for error in errors {
            write!(f, "\n- {}", error);
        }
    }
}
```